src/components/PageViews/ArticleViewsGraph.jsx

```diff
@@ -61,7 +62,13 @@ const ArticleViewsGraph = ({ articles }) => {
     return (
         <div className='graph w-full flex flex-col items-center'>
             <h2 className='font-bold text-white text-xl self-start'>Views for Multiple Articles</h2>
-            {loading ? <Loading /> : viewData.labels ? <Line data={viewData} /> : <p>No data available.</p>}
+            {loading ? (
+                <Loading />
+            ) : viewData.labels ? (
+                <Line data={viewData} />
+            ) : (
+                <p className='text-white text-2xl absolute top-1/2 right-1/4 transform -translate-x-1/2 -translate-y-1/2'>No data available.</p>
+            )}
```

Comment:
It is a good start @siddharthisrani !
It seems to work the first time, but not for subsequent tries.

Try the following :

I have noticed some issues with the `<ArticleViewsGraph/>` component:

1. Set the default value of loading to false :
```typescript
    const [loading, setLoading] = useState(false);
```

2. Then set it to true when a new API request is made 

```typescript
     useEffect(() => {
        const fetchAllViews = async () => {
            try {

               setLoading(true);  // <------

                const allDataPromises = articles.map(async (article) => {
                    const response = await fetchPageViewsCount(article);
                    const views = response.items;
                    const dates = views.map((view) => {
                        const year = view.timestamp.substring(0, 4);
                        const month = view.timestamp.substring(4, 6);
                        const day = view.timestamp.substring(6, 8);
                        const dateString = `${year}-${month}-${day}`;
                        const date = new Date(dateString);
                        return date.toLocaleDateString();
                    });
                    const counts = views.map((view) => view.views);
                    return { article: article.article, dates, counts };
                });
                .........
```
3. Add a `finally` block in which you will set the loading state to false : (Whether the request succeeds or failed, we need to make sure the loading is set to false so we can handle the error properly)

4. In the `catch` block : Set data to `{}`

Reply:
Hii @CREDO23 ,
Sure, I'll make those changes and update the PR accordingly. Thank you for the feedback!

I'll work on setting the default value of loading to false, updating it to true when a new API request is made, and adding a finally block to set loading back to false. I'll also handle setting the data to an empty object in the catch block.

I'll let you know once the changes are done.

Reply:
Perfect @siddharthisrani !